### ISSUE TITLE
Uninstalling VS Code takes a long time, when enabling the Windows 11 context menu action

### ISSUE THREAD

**Issue by dbaeumer** [microsoft/vscode#164689]
Testing #164465

When uninstalling VS Code with enabled Open menu entry the uninstaller seems to hang at the beginning and it takes quite some time until it finishes. I never experienced this when testing the final Windows bits during endgame.

**Comment by deepak1556** [comment]
We now run a powershell command during uninstall to remove the appx package and this can take longer if file explorer is actively holding reference to the COM object. I don't have ideas now to make this better since this flow happens outside the application and not much we can intervene in this flow.

**Comment by joaomoreno** [comment]
How long approximately?

**Comment by joaomoreno** [comment]
@dbaeumer Is this still an issue?

**Comment by dbaeumer** [comment]
Yes, still repros for me.

**Comment by Henning742** [comment]
Would running that PS command only after the next OS restart help?

**Comment by deepak1556** [comment]
I was unable to repro with latest insiders but not sure if it has to do with canary channel of windows I am using, @dbaeumer @joaomoreno can you help confirm if this is still an issue. Thanks!

**Comment by dbaeumer** [comment]
I can still reproduce the hang in a Windows Sandbox.

**Comment by deepak1556** [comment]
@dbaeumer thanks for confirming, what is the version of your OS ?

**Comment by dbaeumer** [comment]
<img width="369" alt="image" src="https://github.com/microsoft/vscode/assets/1931590/e6d61e00-0359-4614-a84a-88c12798af31">

**Comment by deepak1556** [comment]
@dbaeumer can you capture a profile following the steps below, it would help moving this issue forward https://github.com/microsoft/vscode/issues/204696

* Open `Windows Performance Recorder` from the OS start menu, if you don't have it installed set it up following the instructions [here](https://learn.microsoft.com/en-us/windows-hardware/test/wpt/windows-performance-recorder#where-to-get-windows-performance-recorder)
* Download the following profile [CloudFileExplorerLogs2.wprp.zip](https://github.com/microsoft/vscode/files/14206831/CloudFileExplorerLogs2.wprp.zip) and add it to the profiles section of windows performance recorder
* Have the settings in windows performance recorder configured as shown in the screenshot below ![wpr_setup](https://github.com/microsoft/vscode/assets/964386/fe48ed2a-35fa-4639-9f5f-28b36c7a8c65)
* Start the recorder and perform the uninstallation steps of VS Code insiders
* Wait for the uninstallation to complete
* Stop the recorder and save the generated etl trace
* Send the trace to Deepak.Mohan@microsoft.com, **do not attach** the trace to this github issue as it will contain PII

**Comment by reves** [comment]
@deepak1556 I checked out the #187015 that tried to fix this issue, and [your comment](https://github.com/microsoft/vscode/pull/187015#issuecomment-1680118560) that justifies the use of double `{{`.

So that's why i'm interested if the `ScriptBlock` on [this line](https://github.com/microsoft/vscode/blob/main/build/win32/code.iss#L107) shouldn't end with double `}}` as well?
Like this: `{{Remove-AppxPackage -Package ""{#AppxPackageFullname}""}}`

No idea if it's important, i just hope it helps.

**Comment by dbaeumer** [comment]
Tried to install the profile and get this:

<img width="542" alt="image" src="https://github.com/microsoft/vscode/assets/1931590/0b842a93-659a-408f-b8aa-58238509bbc3">

**Comment by dbaeumer** [comment]
Any tips?

**Comment by deepak1556** [comment]
Sorry, try with this new profile [CloudFileExplorerLogs.zip](https://github.com/microsoft/vscode/files/14219882/CloudFileExplorerLogs.zip)

If import was successful, you should see the following entry 
![Screenshot 2024-02-09 181900](https://github.com/microsoft/vscode/assets/964386/27bff2c1-16fc-410b-87ad-e944a1d8cb65)

**Comment by dbaeumer** [comment]
@deepak1556 still the same problem with the new profile

![cast](https://github.com/microsoft/vscode/assets/1931590/96b0c24d-d603-4a9f-900b-01badb7ca876)

**Comment by deepak1556** [comment]
Thanks for trying it out, I have reached out to the file explorer team, will follow up.

**Comment by Vaibhav-Naik-1** [comment]
This feature is not been added because
> When uninstalling VS Code with enabled Open menu entry the uninstaller seems to hang at the beginning and it takes quite some time until it finishes. I never experienced this when testing the final Windows bits during endgame.

So I request developers to let the us (the users) to decide whether we want this option or not when installing the `Visual Studi Code`, that do we want it on the Windows 11 Primary/Main Context Menu or not.
If we want we will check the box or else we won't and by default keep it unchecked.
![Provide an option here](https://github.com/microsoft/vscode/assets/94748147/4b449487-5b67-4169-9d3f-178622d9e5d7)

**Comment by maisondasilva** [comment]
I was unable to reproduce in Beta, Dev and Canary and also in Sandbox.

**Comment by albertvaka** [comment]
Should this be a blocker? The lack of an entry in the context menu affects way more users than this potential issue.

**Comment by EuDs63** [comment]
> Should this be a blocker? The lack of an entry in the context menu affects way more users than this potential issue.

I couldn't agree more! For me, I will probably never uninstall it once I have downloaded it. However, I need this feature to save time.

**Comment by enginelesscc** [comment]
In my case on one of my machines, this is caused by powershell.exe repeatedly crashing in a loop and it takes at least 2 minutes until the uninstaller progresses.

Noticed this in 2 ways:
- TaskManager shows werfault spawning alot during uninstall process
- I enabled windows-7-style "<program> has stopped working" popups (HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\Windows Error Reporting\DontShowUI = 0)

The uninstaller should probably handle this case to avoid stalling forever

**Comment by thernstig** [comment]
@deepak1556 is the reason this is not progressing because of this?

> I have reached out to the file explorer team, will follow up.

I think everyone waiting could use an update on what happened there. Maybe that will silence some questions?

**Comment by calvinwonghk** [comment]
yes still there on 1.99 and during uninstall it hangs and unable to finish !!

![Image](https://github.com/user-attachments/assets/54c78df7-e58c-497c-bf86-20b12936b467)

**Comment by BanditTech** [comment]
Having to wait for uninstall for some fringe cases would be tolerable compared to not having the option directly in the menu for all users.
I just tested if there is the issue on attempting to uninstall and there was no issue at all uninstalling.
